### ISSUE TITLE
Inner symbols should be nested in structure panel

### ISSUE THREAD

**Issue by jfehrle** [reasonml-editor/reasonml-idea-plugin#73]
plugin version: 0.57

### Description
Inner symbols, such as symbols that are defined inside functions, should appear in subtrees under the outer function rather than at the top level of the tree.  This gets particularly confusing for commonly-repeated variable names such as "id" or "e"--the only way to tell where they are in the source is to click on them.

For example, in the screen shot, I just clicked on the highlighted `id` line, which is for a symbol defined inside `merge_proof_branch`.  There are multiple symbols with this same name.  This `id` should be a subnode of `merge_proof_branch`.  (`merge_proof_branch` is currently a leaf node.)

![image](https://user-images.githubusercontent.com/1253341/42414686-2c97ae1e-81ef-11e8-837b-12235fce007e.png)

**Comment by giraud** [comment]
could you insert a sample code that is not working here please ? because this is a parser issue and I need the root cause.

**Comment by jfehrle** [comment]
I'll provide sample code shortly.  However, it is interesting that the `id` in my screen shot is no longer listed in the structure panel (but there are still 4 `id`s for nested let's.

**Comment by jfehrle** [comment]
Here is a reduced example that may not compile plus the entire original file here: [stm.ml.gz](https://github.com/reasonml-editor/reasonml-idea-plugin/files/2189164/stm.ml.gz)
.  Notice that the last `id` doesn't show up in the structure panel--another bug.

```

let process_transaction ~doc ?(newtip=Stateid.fresh ())
      match c with
      (* Meta *)
      | VtMeta, _ ->
        let id, w = Backtrack.undo_vernac_classifier expr ~doc in

      (* Query *)
      | VtQuery, w ->
          let id = VCS.new_node ~id:newtip () in

          let id = VCS.new_node ~id:newtip () in

let fun2 () =
  let id = 123 in  (* THIS ONE DOESN"T APPEAR IN THE STRUCTURE PANEL *)
  Printf.printf ""
```

**Comment by jfehrle** [comment]
Also, `_` should not be listed in the structure panel:

![image](https://user-images.githubusercontent.com/1253341/42648681-6c20fa56-85bc-11e8-8e09-6ba3ba8595a8.png)

**Comment by giraud** [comment]
you should not see `let _` anymore in v0.59

**Comment by giraud** [comment]
should be fixed in 0.60.1
